extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProcessor.java

```diff
@@ -47,6 +48,12 @@ void additionalProviders(BuildProducer<AdditionalBeanBuildItem> additionalBean,
         additionalReaders
                 .produce(new MessageBodyReaderBuildItem(JacksonMessageBodyReader.class.getName(), Object.class.getName(),
                         Collections.singletonList(MediaType.APPLICATION_JSON)));
+        additionalReaders
+                .produce(new MessageBodyReaderBuildItem(JacksonMessageBodyReader.class.getName(), Collection.class.getName(),
```

Comment:
We need `Collection`, because when checking interfaces, `Object` (as a "super-class") doesn't get checked (I am pretty sure that it's spec mandated).
By adding `Collection`, all collections will work just fine.

Reply:
The spec mandates that we don't check `Object` for interface types? That'd be insane. I can't believe that.

Reply:
My theory is that in code like `Serialisers.findReaders` we should special case something like this:

```java
    public List<MessageBodyReader<?>> findReaders(ConfigurationImpl configuration, Class<?> entityType,
            MediaType mediaType, RuntimeType runtimeType) {
        List<MediaType> mt = Collections.singletonList(mediaType);
        List<MessageBodyReader<?>> ret = new ArrayList<>();
        Deque<Class<?>> toProcess = new LinkedList<>();
        Class<?> klass = entityType;
        if (primitivesToWrappers.containsKey(klass))
            klass = primitivesToWrappers.get(klass);
        QuarkusMultivaluedMap<Class<?>, ResourceReader> readers;
        if (configuration != null && !configuration.getResourceReaders().isEmpty()) {
            readers = new QuarkusMultivaluedHashMap<>();
            readers.putAll(this.readers);
            readers.addAll(configuration.getResourceReaders());
        } else {
            readers = this.readers;
        }
        do {
            Collections.addAll(toProcess, klass.getInterfaces());
            if (klass == Object.class || klass.getSuperclass() == null) {
                //spec extension, look for interfaces as well
                //we match interfaces before Object
                Set<Class<?>> seen = new HashSet<>(toProcess);
                while (!toProcess.isEmpty()) {
                    Class<?> iface = toProcess.poll();
                    List<ResourceReader> goodTypeReaders = readers.get(iface);
                    readerLookup(mediaType, runtimeType, mt, ret, goodTypeReaders);
                    for (Class<?> i : iface.getInterfaces()) {
                        if (!seen.contains(i)) {
                            seen.add(i);
                            toProcess.add(i);
                        }
                    }
                }
            }
            List<ResourceReader> goodTypeReaders = readers.get(klass);
            readerLookup(mediaType, runtimeType, mt, ret, goodTypeReaders);
// if we're looking for an interface, pretend it extends Object to make it match,
// we will still match superinterfaces before we match Object so it will work
if(klass.isInterface() {
 klass = Object.class;
} else {
            klass = klass.getSuperclass();
}
        } while (klass != null);

        return ret;
    }

```

Reply:
I do think the interface thing was a special in the spec, but I need to check.

Reply:
So it seems like the spec doesn't mention anything. Moreoever, we do that same trick you mentioned for writers, so I updated the PR to use it for readers as well